Common/CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+# This will build the Common library
+#
+# On Linux,
+#   - We need to link to the `librt.so`, and for some reason, CMake cannot find it
+#     easily, so, Libraries module tries to find it and provides it to Common here.
+#   - Since we are not using Conan, I link to the libjsoncpp, using the PkgConfig variables
+#
+#
+list(APPEND CMAKE_MESSAGE_CONTEXT Common)
+
+configure_file(appinfo.cpp.in appinfo.cpp)
+message(STATUS "appinfo.cpp is successfully generated...")
+
+file(GLOB_RECURSE HEADER_FILES "${CMAKE_CURRENT_LIST_DIR}/*.h")
+
+file(GLOB_RECURSE SOURCE_FILES "${CMAKE_CURRENT_LIST_DIR}/*.cpp")
+list(APPEND SOURCE_FILES ${CMAKE_CURRENT_BINARY_DIR}/appinfo.cpp)
+
+add_library(Common ${HEADER_FILES} ${SOURCE_FILES})
+
+if(LINUX)
+
+  target_link_libraries(Common PUBLIC ${_LIB_RT})
+  target_link_options(Common PUBLIC -lrt)
+
+endif()
+
+target_include_directories(
+  Common
+  PUBLIC # JASP
+         jaspColumnEncoder
+         # R
+         ${R_INCLUDE_PATH}
+         ${R_HOME_PATH}/include
+         ${RCPP_PATH}/include
+         #
+         $<$<PLATFORM_ID:Linux>:${_PKGCONFIG_LIB_JSONCPP_INCLUDEDIR}>)
+
+target_link_libraries(
+  Common
+  PUBLIC # jsoncpp
+         $<$<BOOL:${USE_CONAN}>:jsoncpp::jsoncpp>
+         $<$<PLATFORM_ID:Linux>:${_PKGCONFIG_LIB_JSONCPP_LIBRARIES}>
+         #
+         LibArchive::LibArchive
+         # Boost
+         Boost::filesystem
+         Boost::system
+         Boost::date_time
```

Comment:
so, `date_time, timer and chrono` used to be dependent on `JASPTIMER_USED` and the same define would make the profiling macros into `noop` or actual profiling depending on that.
See `Common/timers.h` 

I guess there is no real harm in having them linked though, those few bytes are probably not noticeable.

Reply:
It definitely doesn't hurt, and later I can fine tune these as we go.